### PR TITLE
check: wait for progress bar output

### LINKED ISSUE #3344: Typo in check command
Just FYI may be missing a space in restic check output.  The line is "no errors were found49 snapshots"

````bash
# restic version
restic 0.12.0 compiled with go1.15.8 on linux/amd64
````
````bash
# restic check
using temporary cache in /tmp/restic-check-cache-358669417
repository 2601de52 opened successfully, password is correct
created new cache in /tmp/restic-check-cache-358669417
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
no errors were found49 snapshots
[0:51] 100.00%  49 / 49 snapshots
````

### PR DESCRIPTION
**Pull request by mathstuf**
Further code will also output to the terminal and the bar's cursor
positioning causes its output to overlap with the remaining output in a
racy way.

Fixes: #3344

What does this PR change? What problem does it solve?
-----------------------------------------------------

Waits for the progress bar to be complete before waiting for other output.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

#3344

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] ~~I have added tests for all code changes.~~ Not sure how to test a race condition.
- [ ] ~~I have added documentation for relevant changes (in the manual).~~
- [ ] ~~There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).~~
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

**Comment by mathstuf** [comment]
Hmm. Apparently `go build` isn't actually building anything and has not output, so…not actually sure this compiles yet.

File: cmd/restic/cmd_check.go [hunk 1]
```diff
@@ -252,12 +253,19 @@ func runCheck(opts CheckOptions, gopts GlobalOptions, args []string) error {
...
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
 		bar := newProgressMax(!gopts.Quiet, 0, "snapshots")
 		defer bar.Done()
 		chkr.Structure(gopts.ctx, bar, errChan)
 	}()
 
+	// Wait for the progress bar to be complete before printing more below.
+	wg.Wait()
```
Comment (MichaelEischer) [inline cmd/restic/cmd_check.go:267]: This will deadlock if the check outputs errors. The correct location is after the `for err := range errChan {`-loop.

**Comment by MichaelEischer** [comment]
`go build ./cmd/restic` also works to build a binary.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**